utilities/release-osci.py

```diff
@@ -0,0 +1,277 @@
+                f"[WRITE] would modify references in stackrox-collector-{self.version_str}.yaml"
+            )
+            return
+
+        print(
+            f"[WRITE] modifying references in stackrox-collector-{self.version_str}.yaml"
+        )
+
+        with open(release_config) as config_file:
+            cfg = yaml.load(config_file, Loader=yaml.Loader)
+
+            # remote image promotion
+            del cfg["promotion"]
+
+            # remove all the tests not marked as post-submit
+            cfg["tests"] = [
+                test for test in cfg["tests"] if test.get("postsubmit", False)
+            ]
+
+            for test in cfg["tests"]:
+                if "steps" not in test:
+                    continue
+
+                if "workflow" in test["steps"]:
+                    workflow = test["steps"]["workflow"]
+                    workflow = workflow.replace(
+                        "stackrox-collector-", f"stackrox-collector-{self.version_str}-"
+                    )
+                    test["steps"]["workflow"] = workflow
+
+            cfg_data = yaml.dump(cfg, Dumper=yaml.Dumper)
+
+        # dump the config back to disk
+        with open(release_config, "w+") as config_file:
+            config_file.write(cfg_data)
+
+    def copy_steps(self):
+        """
+        Copies all step-registry files into a release directory, to scope
+        them off from the usual 'master' step-registry. Most of the files need
+        to be renamed, and references to each other within the files need to
+        be changed as well.
+
+        The renaming is based on the directory structure relative to
+        ci-operator/step-registry.
+
+        e.g. ci-operator/step-registry/stackrox/collector/integration-tests/stackrox-collector-integration-tests-workflow.yaml
+
+             becomes
+
+            ci-operator/step-registry/stackrox/collector/release-<version>/integration-tests/stackrox-collector-release-<version>-integration-tests-workflow.yaml
+        """
+        self.mkdir(self.step_from_root(self.version_str))
+
+        # get all top-level steps that aren't release directories
+        subdirs = next(os.walk(self.from_root(g_steps_root)))[1]
+        subdirs = filter(lambda d: "release" not in d, subdirs)
+
+        for subdir in subdirs:
+            self.mkdir(self.step_from_root(self.version_str, subdir))
+
+            for root, dirs, files in os.walk(self.step_from_root(subdir)):
+                for dir in dirs:
+                    #
+                    # we need to reconstruct the path to include the release
+                    # directory, so using relpath we can get the part of the
+                    # path relative to ci-operator/step-registry/stackrox/collector
+                    # and then tack it onto the end of the release directory
+                    #
+                    src = os.path.join(root, dir)
+                    rela = os.path.relpath(src, self.from_root(g_steps_root))
```

Comment:
Is `rela` a typo for `real` or short for `relative`?

Reply:
it's short for relative - I'll make this clearer